src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -17,83 +27,220 @@
 
     private static final Logger log = Logger.getLogger();
 
-    private String searchServiceHost;
+    private static final String ERROR_DELETE_DOCUMENT =
+            "Failed to delete document(s) %s in Solr. Root cause: %s ";
+    private static final String ERROR_SEARCH_DOCUMENT =
+            "Failed to search for document(s) %s from Solr. Root cause: %s ";
+    private static final String ERROR_SEARCH_NOT_IMPLEMENTED =
+            "Search service is not implemented";
+    private static final String ERROR_PUT_DOCUMENT =
+            "Failed to put document(s) %s into Solr. Root cause: %s ";
+    private static final String STUDENT_COLLECTION_NAME = "students";
+    private static final String INSTRUCTOR_COLLECTION_NAME = "instructors";
 
-    public SearchManager(String searchServiceHost) {
-        this.searchServiceHost = searchServiceHost;
-    }
+    private static final int NUM_OF_RESULTS = 20;
 
-    private boolean isSearchServiceActive() {
-        return !StringHelper.isEmpty(searchServiceHost);
+    private HttpSolrClient client;
+
+    public SearchManager(String searchServiceHost) {
+        if (!StringHelper.isEmpty(searchServiceHost)) {
+            this.client = new HttpSolrClient.Builder(searchServiceHost).build();
+        }
     }
 
     /**
      * Searches for students.
      *
      * @param instructors the constraint that restricts the search result
      */
-    public StudentSearchResultBundle searchStudents(String queryString, List<InstructorAttributes> instructors)
+    public List<StudentAttributes> searchStudents(String queryString, List<InstructorAttributes> instructors)
             throws SearchNotImplementedException {
-        if (!isSearchServiceActive()) {
+        if (client == null) {
             throw new SearchNotImplementedException();
         }
-        // TODO
-        throw new SearchNotImplementedException();
+
+        QueryResponse response = null;
+        SolrQuery studentQuery = new SolrQuery();
+
+        String cleanQueryString = escapeSpecialChars(queryString);
+        studentQuery.setQuery("\"" + cleanQueryString + "\"");
```

Comment:
Not advisable to put quotes in. This means we implicitly treat the entire search term as exact match, but that's not always the user's intention.

Reply:
True. As I know, the original search API does tokenise query string by whitespace.

One thing different for Solr is:

> The "@" character is among the set of token-splitting punctuation, so email addresses are not preserved as single tokens.

This leads to irrelevant search results for emails, we might need to specially handle it in some way.